externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala

```diff
@@ -36,6 +37,14 @@ object KDFRegistry {
     "string",
     "1.3.0")
 
+  val engine_name: KyuubiDefinedFunction = create(
+    "engine_name",
+    udf(() => SparkContext.getOrCreate().getConf.get("spark.app.name")).asNonNullable(),
```

Comment:
SparkSession.active.sparkContext.appName

Reply:
Thanks, I will update it later.

Reply:
'SparkSession.active' can't get the SparkSession when it is running in the executor, so the 'SparkSession.active.sparkContext.appName' have an exception of 'No active or default Spark session found.'

Reply:
How about `SparkEnv.get.conf`?

Reply:
SparkEnv.get.conf can work well,  what is the difference between SparkEnv.get and SparkContext.getOrCreate().getConf, It looks like that their output is the same about appName.

Reply:
SparkContext only existing in driver.

For sparkEnv, it is about seria/lizerManager/mapOutputTracker/blockManager.
And there are driverEnv and executorEnv.
So, you can access it both on driver and executor.

Reply:
FYI:
https://github.com/apache/spark/blob/554d5fef13579cefea083e69b98bce32c0c99126/core/src/main/scala/org/apache/spark/SparkEnv.scala#L197

Reply:
Thanks, I will update it later.

Reply:
can we get `val appName = SparkSession.active.sparkContext.appName` at driver side ahead and then pass  the `appName` into the udf?

Reply:
Make sense, getting it directly is more concise unless there are some advantages.